### PR TITLE
Centralize the secret env schema

### LINKED ISSUE #1485: Duplication: '<NAME>_DOMAINS' secret-schema walk + CSV split re-implemented in 3 TS packages (node-server, chrome-extension, cloud-core) with 2 different domain splitters
The `NAME=value` / `NAME_DOMAINS=csv` companion-key convention that pairs a secret with its authorized-host allowlist is the wire contract between the in-shell `secret` command, node-server's file store, the chrome-extension SW's fetch-proxy backend, and cloud-core's resume pipeline. It is documented explicitly as a shared schema (see `packages/chrome-extension/src/secrets-storage.ts:9-14` — *"Storage schema mirrors the in-shell secret command and the SW mount sign-and-forward handler"*) — but every implementation re-derives it inline from scratch, and the domain-CSV splitter has already drifted.

## Copies

| # | File | Notes |
|---|---|---|
| A | `packages/node-server/src/secrets/env-secret-store.ts:16, 30, 46-49, 56, 66-71, 104-108` | `DOMAINS_SUFFIX` const + `parseDomains(value)` splitter + inline `endsWith(DOMAINS_SUFFIX)` filter in `list()`. |
| B | `packages/chrome-extension/src/secrets-storage.ts:17, 42-51, 64-67, 71` | Same `DOMAINS_SUFFIX`, its own inline splitter in `listSecrets`, own `deleteSecret`/`setSecret` pair-write. |
| C | `packages/cloud-core/src/operations/resume.ts:47-72` | Inline `parseSecretsEnv(text)` that does the identical two-pass `_DOMAINS` walk with a hand-written slice `key.slice(0, -'_DOMAINS'.length)` and its own splitter. |
| D (parity target) | `packages/swift-server/Sources/Keychain/EnvFileFormat.swift:10, 63-64, 88-94, 103-107` | Swift port; `private let domainsSuffix = \"_DOMAINS\"` + `parseDomains(raw)`. Comment on line 14 already flags this as *"Mirrors packages/node-server/src/secrets/env-file.ts so the same blob can round-trip"* — the TS side, however, has 3 copies to mirror. |

Plus test fixtures with the `_DOMAINS` literal in `packages/node-server/tests/**` (~10 files) and `packages/cloud-core/tests/**` (~5 files) that all encode the wire format.

## Concrete drift already present

The three TypeScript splitters do NOT agree on how to parse a comma-list:

- `env-secret-store.ts:106-108` — `.split(',').map(d => d.trim()).filter(d => d.length > 0)` (trims + drops empties)
- `secrets-storage.ts:47-50` — `.split(',').map(d => d.trim()).filter(Boolean)` (trims + falsy filter)
- `resume.ts:56-60` — `.split(',').filter(Boolean)` (**no trim**)

Given a `MY_KEY_DOMAINS=api.github.com, *.github.com` line (a natural CSV with a post-comma space that a human editing `~/.slicc/secrets.env` or a hand-authored `--env-file` will produce), the CLI store yields `['api.github.com', '*.github.com']` but cloud-core's `parseSecretsEnv` yields `['api.github.com', ' *.github.com']`. The leading space breaks downstream wildcard matching against the request URL — the second domain silently authorizes nothing. This path runs on every `slicc --cloud resume` when a `refreshSecretsContents` payload is applied, and on the worker-side resume in `cloudflare-worker/src/cloud/cloud-sessions-do.ts`.

Secondary drift: the extension has S3-profile validation (`PROFILE_RE`, `deriveS3Domains`, `validateS3ProfileInput` in `secrets-storage.ts:20, 84-96, 116-136`) with a comment claiming it *"matches the server-side regex in sign-and-forward.ts"*; node-server's env-secret store enforces no analogous constraint, so a key that the extension would reject up-front can be written from the CLI and then read back by both.

## Why this is impactful

- **4+ copies of the same schema walk** across three TypeScript packages (five with the tests) plus a Swift mirror; the schema is defined by cross-consumer convention with no single owner.
- **Already-observed semantic drift** in a security-adjacent code path (host allowlist for the credential-injecting fetch proxy).
- **`@slicc/shared-ts` already exists for exactly this** — `secret-masking.ts`, `secrets-pipeline.ts`, `oauth-extra-domains-storage.ts`, and `session-secret-store.ts` were moved there for the same reason, and the Swift `EnvFileFormat.parse` mirror already sits in the byte-for-byte parity harness (`Tests/CrossImplementationTests.swift` → `packages/shared-ts/tests/cross-impl-vectors.test.ts`).
- Not covered by prior closed duplication issues (#889, #991, #1087, #1213, #1310, #1294, #1332) — those addressed SigV4, fetch-proxy sync/async, base64 codecs, `/electron` path, RPC bridges, wire types, and hosted-UI architecture respectively.

## Suggested remediation

Move the `_DOMAINS` companion schema to `@slicc/shared-ts/src/secret-env-schema.ts`:

- `export const DOMAINS_SUFFIX = '_DOMAINS';`
- `export function parseDomainsCsv(raw: string): string[]` — one canonical splitter (trim + drop empties).
- `export function pairEnvEntriesToSecrets(entries: EnvEntry[]): SecretEntry[]` — the two-pass walk used by node-server `list()`, cloud-core `parseSecretsEnv`, and (adapted for `chrome.storage.local`) extension `listSecrets`.
- `export const PROFILE_RE`, `deriveS3Domains`, `validateS3ProfileInput` — the S3-profile helpers currently extension-only.

Then rewrite:
- `packages/node-server/src/secrets/env-secret-store.ts` to import the splitter + walker.
- `packages/chrome-extension/src/secrets-storage.ts` to import `DOMAINS_SUFFIX` + splitter (walker adapts to `StorageArea`).
- `packages/cloud-core/src/operations/resume.ts::parseSecretsEnv` → drop; import `pairEnvEntriesToSecrets` (reuse `parseEnvFile` from `@slicc/shared-ts` — move it there too, since Swift's `EnvFileFormat.parse` already pins the format).
- Update `Sources/Keychain/EnvFileFormat.swift`'s doc comment to point at the shared-ts owner.

Add a `packages/shared-ts/tests/secret-env-schema.test.ts` pinning the CSV corner cases (`\"a, b\"`, `\"a,\"`, `\"a,,b\"`, whitespace-only) so no consumer can re-drift.

### PR DESCRIPTION
**Pull request by trieloff**
Closes #1485

## Solution
Makes `@slicc/shared-ts` the canonical owner of env parsing, `_DOMAINS` pairing, CSV normalization, and S3 profile helpers.

## Misc
All TypeScript coverage gates and builds passed; the Swift build was unavailable because this VM has no Swift toolchain.

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) | [View session](https://cosmos.augmentcode.com/session?agentId=01KX9WY31TZ024Y80NB6GCBNFX&panel=chat)

### CODE REVIEW

**Comment by trieloff** [comment]
<sup>[**PR Author Agent**](https://cosmos.augmentcode.com/session?agentId=01KX9WY31TZ024Y80NB6GCBNFX)⚡</sup>

👋 I've got this PR.

I'll handle review feedback, CI failures, and merge conflicts, and ping you the moment it's ready for review. Drop a comment anytime.

Marking it ready and picking reviewers are your call — I'll leave both alone.

File: packages/cloud-core/src/operations/resume.ts [hunk 1]
```diff
@@ -96,10 +69,14 @@ export function applyConeConfigDelta(
     base = validateConeConfig({
       model: parsed.model ?? DEFAULT_MODEL,
       accounts: parsed.accounts ?? [],
-      secrets: parseSecretsEnv(secretsEnv),
+      secrets: pairEnvEntriesToSecrets(parseEnvFile(secretsEnv)),
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/cloud-core/src/operations/resume.ts]: <sup>[**PR Author Agent**](https://cosmos.augmentcode.com/session?agentId=01KX9WY31TZ024Y80NB6GCBNFX)⚡</sup>

Fixed in `84c516bc`: cloud resume now uses the shared value-preserving parser, keeping quoted and whitespace-padded secrets verbatim while still normalizing domain CSVs. Added shared-parser and cloud round-trip regressions; both package coverage gates pass.
